.github/scripts/gen-matrix.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+#
+# This script is used to create the matrix for the language specific images, for example in the `debian-sdk` job. The
+# created matrix has no variables, instead we list each desired combination explicitly in the `include` field.
+# https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-adding-configurations
+#
+# matrix = {
+#     "include": [
+#         {"sdk": "go",     "arch": "amd64", "language_version": "1.21.1", "default": True},
+#         {"sdk": "go",     "arch": "arm64", "language_version": "1.21.1", "default": True},
+#         {"sdk": "python", "arch": "amd64", "language_version": "3.9",    "default": True,  "suffix": "-3.9"},
+#         {"sdk": "python", "arch": "arm64", "language_version": "3.9",    "default": True,  "suffix": "-3.9"},
+#         {"sdk": "python", "arch": "amd64", "language_version": "3.10",   "default": False, "suffix": "-3.10"},
+#         {"sdk": "python", "arch": "arm64", "language_version": "3.10",   "default": False, "suffix": "-3.10"},
+#         ...
+#     ]
+# }
+#
+#  * `language_version` is the version of the language runtime to use, for example `3.9` for Python.
+#  * `suffix` is an optional suffix to append to the image name, for example `-3.9` to generate `pulumi-python-3.9`.
+#  * `default` indicates that this is the default language_version. We will push two tags for the image, once
+#     with and once without the suffix in the name, for example `pulumi-python-3.9` and `pulumi-python`.
+#
+# When running this script, pass `--no-arch` to exclude the `arch` field from the matrix. This is used in the release
+# job for creating the docker manifests. For example the manifest for `pulumi-python:3.123.0-debian` includes the
+# images `pulumi-python-3.123.0-debian-amd64` and `pulumi-python-3.123.0-debian-arm64`, so we don't need to iterate
+# over the architecutres in the matrix.
+#
+import json
+import sys
+
+INCLUDE_ARCH = False if len(sys.argv) > 1 and sys.argv[1] == "--no-arch" else True
+
+archs = ["amd64", "arm64"] if INCLUDE_ARCH else [None]
+matrix = {"include": []}
+
+# SDKs without version suffixes
+sdks = {
+    "nodejs": "18",
+    "go": "1.21.1",
+    "dotnet": "6.0",
+    "java": "not-used-yet",
+}
+# Python versions
+python_default_version = "3.9"
+python_additional_versions = ["3.10", "3.11", "3.12"]
```

Comment:
> The image without suffix `pulumi/pulumi-python` continues to use 3.9.

This makes sense to me. I assume at some point we'll want to make the default version use a more recent version, either the latest (or possibly latest-1). How would we go about doing that? Announcing the plan/timeline and then doing it? Would it look like the following (assuming we pick 3.12 as the default)?

```python
python_default_version = "3.12"
python_additional_versions = ["3.9", "3.10", "3.11"]
```

Reply:
Yes, I might revisit how we do the version list and default version setting when I add the matrix for nodejs, but for now this is how it works.

It's going to be a breaking change, but one I think we need to do. Users that require 3.9 can switch to the `python-3.9` image. The announcement should also clearly set the expectation that the `python` image will be updated regularly with a more recent python version.